### PR TITLE
Shivani | Test | Fetch Go cardless integrated bank balance in dashboard page

### PR DESCRIPTION
**Pull request by shivanivani02**
https://app.clickup.com/t/86cx4gg7h

## Summary by CodeRabbit

- **New Features**
  - Enhanced display logic and interactivity for bank account details.
  - Improved connection indicators for bank accounts.
  - Increased the number of bank accounts fetched in API calls.
  - Updated dialog behavior to prevent closing on outside clicks.

- **Bug Fixes**
  - Streamlined logic for the "connect bank" button visibility and functionality.

- **Documentation**
  - Updated bindings for dropdown component to utilize reactive data sources.

- **Refactor**
  - Improved state management and lifecycle handling in the bank link component.

- **Chores**
  - Adjusted styling for conditional height in the bank account component.

### CODE REVIEW

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss [hunk 1]
```diff
@@ -26,6 +26,11 @@ a.connect-bank {
     cdk-virtual-scroll-viewport {
         height: 270px;
     }
+    &.conditional-height {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss]: remove

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html [hunk 2]
```diff
@@ -11,14 +11,19 @@
             </div>
         </div>
         <mat-card-content class="banks-scrollbar-wrapper" *ngIf="localeData">
-            <div class="bank-details-block">
+            <div class="bank-details-block" [ngClass]="{ 'conditional-height': isGocardlessSupportedCountry && hasIntegrationScope && bankAccounts?.length > 0 }">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html]:
```suggestion
            <div class="bank-details-block">
```

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss [hunk 3]
```diff
@@ -26,6 +26,11 @@ a.connect-bank {
     cdk-virtual-scroll-viewport {
         height: 270px;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss]:
```suggestion
        height: 334px;
```

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss [hunk 4]
```diff
@@ -64,6 +69,6 @@ p {
     }
 }
 
-.icon-tick {
-    font-size: 14px;
+.icon-plus ,.icon-tick {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.scss]: use class font-12

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html [hunk 5]
```diff
@@ -11,14 +11,19 @@
...
                 <cdk-virtual-scroll-viewport [itemSize]="61" *ngIf="!isLoading && bankAccounts?.length > 0"
                     class="virtual-group-list overflow-x-hidden" tabindex="0">
                     <div class="wrap-single-bank-detail">
                         <div [matTooltip]="bankaccount?.accountBankTransactionTotal?.transactionLastFetchedAt"
                             matTooltipPosition="right" class="single-bank-detail" *cdkVirtualFor="let bankaccount of bankAccounts"
                             >
-                            <p class="bank-name">{{ bankaccount.name }}</p>
+                            <p class="bank-name">{{ bankaccount.name }}
+                            {{bankaccount?.transactionLastFetchedAt}}&nbsp;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html]: check and remove

File: apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html [hunk 6]
```diff
@@ -11,14 +11,19 @@
...
+                                <span *ngIf="!bankaccount.accountBankTransactionTotal && isGocardlessSupportedCountry && hasIntegrationScope && bankAccounts?.length > 0"><i class="icon-plus text-primary cursor-pointer"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/home/components/bank-accounts/bank-accounts.component.html]:
```suggestion
                                <span *ngIf="!bankaccount?.accountBankTransactionTotal && isGocardlessSupportedCountry && hasIntegrationScope"><i class="icon-plus text-primary cursor-pointer"
```

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 7]
```diff
@@ -3192,14 +3195,16 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
-        this.dialog.open(BankLinkComponent, {
+        const dialogRef = this.dialog.open(BankLinkComponent, {
             data: data,
-            panelClass: ['mat-dialog-md']
+            panelClass: ['mat-dialog-md'],
+            disableClose: true
         });
-    }
 
+        dialogRef.afterClosed().pipe(take(1), tap(response => { if (response) this.isBankAccountConnected = true; this.showBankLinkButton = false })).subscribe();
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/ledger/ledger.component.ts]:
```suggestion
        dialogRef.afterClosed().pipe(take(1), tap(response => { if (response) this.isBankAccountConnected = true; this.showBankLinkButton = false; })).subscribe();
```

File: apps/web-giddh/src/app/shared/bank-integration/bank-link/bank-link.component.ts [hunk 8]
```diff
@@ -118,5 +117,21 @@ export class BankLinkComponent implements OnInit {
                 additional: item
             }
         });
+
+        if (this.bankLinks.length === 1) {
+            this.defaultSelectedBank$.next(`${bankList[0].bankName} ****${bankList[0].bankResource?.accountNumber ? bankList[0].bankResource.accountNumber.slice(-4) : 'N/A'}`);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/bank-integration/bank-link/bank-link.component.ts]:
```suggestion
            this.defaultSelectedBank$.next(`${bankList[0]?.bankName} ****${bankList[0]?.bankResource?.accountNumber ? bankList[0]?.bankResource?.accountNumber?.slice(-4) : 'N/A'}`);
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

**Review by dvCodeWorld — APPROVED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

**Pull request merged**